### ISSUE TITLE
Deprecated function finally removed in Python 3.8 causing test fail

### ISSUE THREAD

**Issue by sausix** [llvm/llvm-project#62]
Affects file:
blob/master/libcxx/utils/libcxx/test/target_info.py (2 occurrences)

Python doc:
https://docs.python.org/2/library/platform.html#platform.linux_distribution

Replacement and sample:
https://pypi.org/project/distro/
```
import distro
distro.linux_distribution(full_distribution_name=False)
```

Other affected people in AUR:
https://aur.archlinux.org/packages/libc%2B%2B/

**Comment by androm3da** [comment]
This has also been discussed on the llvm-dev/libcxx-dev lists: https://groups.google.com/forum/#!topic/llvm-dev/32ZwKw8VtnU

It might not be feasible to depend on 3rd party `distro`.  But there's other options: it's only used to filter out SLES, maybe this would be better designed as a `glibc` feature test.

**Comment by sausix** [comment]
Not installing 3rd party dependencies is ok. But maybe it can be used as fallback with try catch instead of running into the python error directly.

But there should at least be an earlier check for the installed Python version.
Or better directly call `python3.7` instead of `python` if found.

Other distributions will use Python 3.8 in future too.

**Comment by androm3da** [comment]
> But there should at least be an earlier check for the installed Python version.

I suppose that could guard it, and we don't even need a fallback.  Since it's only used by SLES11, if Python's too new to support this feature then we just don't emit this feature.

> Other distributions will use Python 3.8 in future too.

Agreed, it's a bug and it's waiting for someone to post a fix.

**Comment by broadwaylamb** [comment]
https://reviews.llvm.org/D72501

**Comment by broadwaylamb** [comment]
Should be fixed by 7b8dc8c57697e95fd0b1248e4494ecc0f929aba1.
